verigreen-collector-impl/src/main/java/com/verigreen/collector/decision/OnFailureHandler.java

```diff
@@ -48,7 +48,19 @@ protected void doHandle() {
                     _commitItem.getParent().getBranchDescriptor().getCommitId(),
                     _commitItem.getMergedBranchName()
                     );
-        }
+        } else if(!_commitItem.getChildCommit().isEmpty()){
```

Comment:
I think there is some room for refactoring here. You have basically the same call to `CollectorApi.getEmailSender().notifyCommiter()` *twice*. 

I think that a cleaner way would be to do this check `!_commitItem.getChildCommit().isEmpty()` within the `else { .. }`  block. And then pass the `null` value in the method call. 

Now I see that you have overloaded the method in `EmailSender.java` and that *takes* care of it. However, I wouldn't recommend using a method overloading pattern just pass a `null` value as an argument. There is really no major change in the overloaded method since you are basically calling the overloaded method. Also, this makes the `EmailSender` API not as clear about its behavior. 

So, do the following. Remove the method in `EmailSender.java` within lines  **68-88** and use the other method to make the call from `OnFailureHandler.java`. 

Does that make sense?